### PR TITLE
Feature #1132 Navbar logo link to home

### LINKED ISSUE #1132: [Feature]: Navbar logo link to home
### Feature Summary

We should make the cuttle logo in the navbar a link

### Detailed Description

We should make the cuttle logo in the navbar a link, this should direct you to home where you can play a game.

Example: You are on `/stats` and click the logo and it brings you to the homepage where you can join a game

### PR DESCRIPTION
**Pull request by IbrahimKhanGH**
## Issue number

Relevant [#1132](https://github.com/cuttle-cards/cuttle/issues/1132)
- Resolves #1132 

## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [x] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [x] Have tests been added to cover any new features or fixes?
  - [x] Has the documentation been updated accordingly?

## Please describe additional details for testing this change
Implementation Details:
- Added router-link wrapper with `display: contents` around logo to maintain existing formatting while enabling navigation to home ('/')

Test Coverage:
- Added tests for authenticated users 
- Tests both desktop (1920x1080) and mobile (iPhone-8) viewports
- Verifies clicking logo navigates from /stats to home (/)

To Test:
1. Run `npm run e2e:gui`
2. Select `navigation.spec.js`
3. Verify logo navigation tests pass in both viewports

### CODE REVIEW

File: src/components/TheHeader.vue
Comment (itsalaidbacklife) [inline src/components/TheHeader.vue:17]: Generally `display: contents` is unideal for links, because we want screen readers to pick up that there is a link tag. Did you find that the layout was messed up without including it? If so we should look to jigger the styles so that the parent can be rendered normally & picked up by screen readers, while allowing it to look the same visually.

If it looks fine without setting the style on the `router-link`, then we can remove the style attribute here without making other changes
```suggestion
          <router-link :to="{ name: ROUTE_NAME_HOME }">
```

**Review by itsalaidbacklife — COMMENTED** [review]
Thank you for your contribution! This is nicely done.

First, re: issue assignment: what you did is correct and generally how we do it. We can't officially assign issues outside of the core team, but you are welcome to pick up any issue you see in the backlog, particularly if it doesn't have a related pull request already associated to it. Adding a comment in the issue communicating that you'd like to work on it (as you've done) further helps reduce conflicts where multiple contributors bump into each other on the same issue.

As for your PR itself, it looks great, tests and all. The only thing I'd like to change is the use of `display: content` on the `router-link` as it's [problematic for accessibility](https://ericwbailey.design/published/display-contents-considered-harmful/). 

If that was added because of specific style wonkery introduced by nesting the image inside another layer, then we'll need some additional styles to account for that, possibly using `position: relative`, or other css on the `<router-link>` as needed. 

If it was "just in case" since in principal we want it to look the same as it did w/o the link wrapper, then we can remove the `style` assignment on the `<router-link>` without making other style changes.

Again, nice work and thank you for your contribution!

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue]: I originally used `display: contents` because the router-link wrapper was pushing the logo down. I was unware of the accessibility issues, so thank you! I went ahead switched to using a logo-container class using position: absolute and transform to maintain the centered positioning. I hope this solution will do, and thank you for the welcome and the opportunity to learn🙏

File: src/components/TheHeader.vue
Comment (itsalaidbacklife) [inline src/components/TheHeader.vue:108]: Is this to move the logo back to the right? It looks like perhaps if we kept the `calc()` call to lower the `left` amount like we previously had it that it might remove the need for this?

File: src/components/TheHeader.vue
Comment (itsalaidbacklife) [inline src/components/TheHeader.vue:105]: Because this is on the same line with the rest of the toolbar content, it would make more sense for this to have display: inline-block.

That said, because this element has `position: absolute`, it's removed from the document flow and the `display` property is not used at all. We shouldn't need this line
```suggestion
```

**Review by itsalaidbacklife — COMMENTED** [review]
This is the right general approach to the css, but there are now additional extraneous changes that we don't want to include. I think the only thing that should be needed is to apply the same css that we previously had on the image to the container itself, without adjusting the transform or left values. If thus goofs up the size or position of the image, we can add additional styles to the image itself to ensure it's positioned relatively with a left: 0 to make sure it starts at the left edge of its container

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue:5]: From what i experimented with, the <v-btn> component automatically applies cursor pointer, so it was redundant

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue:62]: this is what I used to route the logo to the home page 

<router-link 
            **:to="{ name: ROUTE_NAME_HOME }"**
            class="logo-container"
          >

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue:105]: Makes sense, will work on these changes

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue:107]: The code initially was a little hard coded and when I had made some changes, it off centered the logo. I had this solution to have a more dynamic approach, but if it is better to keep the original solution I can do that as well

File: src/components/TheHeader.vue
Comment (itsalaidbacklife) [reply src/components/TheHeader.vue:5]: @IbrahimKhanGH that makes sense; good catch!

Btw one really handy thing you can do when submitting a PR is add PR comments like this one preemptively to explain anything that's not immediately obvious. Things that aren't necessarily "in scope" for an issue but were quick fix like this can be a good fit for a quick comment. Same goes for anything that you expect the reviewer might read and ask "why are they changing this?"

PR comments can be a helpful supplement to code comments when the thing you want to explain isn't about the end result of the code but rather the motivation for the change. Like in this case, there wouldn't be a need to see  any comment in the code after this is merged since we're actually removing the style from this element so there is nothing to comment on. But the change itself is justified by just what you noticed: this style is redundant. That's a good reason for a quick PR note so the reviewer can get quickly up to speed. It also makes for a good chance for you to review your own changes to catch anything that is included by accident and to generally make sure you're PRing exactly the changes you intend.

None of this is a criticism, I just figured that since you mentioned this being your first open source PR that you might like to hear that it's okay to comment on your own PR preemptively and that it can be helpful to do so.

Again, good catch on this extraneous style; we do want to remove those. If it turned out to be a bigger issue changing lots of files to clean something else up, then it would make sense to address it in a separate issue + PR. But since it's just a one line fix, it's fine to include it in your existing one, just as you've done.

File: src/components/TheHeader.vue
Comment (itsalaidbacklife) [reply src/components/TheHeader.vue:106]: It looks good to me visually across devices sizes and isn't wildly complicated so I think this works

**Review by itsalaidbacklife — APPROVED** [review]
Looks good. Thank you for your contribution! Particularly nice work adding tests for the changed behavior. The navigation spec is exactly where those belong and the tests look to correctly describe the intended behavior, confirming the feature's function. Well done!

Feel free to pick up any other issue if you'd like to contribute further to the project. If you'd like to chat with other contributors and team members, you're welcome to join our [discord](https://discord.gg/RHqzRMzds3) where we can discuss finding issues to work on and how to go about them. Thanks again!

File: src/components/TheHeader.vue
Comment (IbrahimKhanGH) [reply src/components/TheHeader.vue:5]: No worries, I want all the feedback I can get. Im completely new to open source, and want to get better at larger scale development, so anything that can teach me to improve is something I want. So, Thank you for all the reviews and the support through my first contribution🙏 I really appreciate it and will be contributing more in the future

**Pull request merged**
